### PR TITLE
6063 els avg calc

### LINKED ISSUE #6063: Core not giving expected equity-like-share and average-entry-valuation values
# Problem encountered
Scenario 001 and 002 in PR #6034 are failing as the `equity-like-share` and `average-entry-valuation` values do not match the expected values calculated following spec [0042](https://github.com/vegaprotocol/specs/blob/master/protocol/0042-LIQF-setting_fees_and_rewarding_lps.md).

Debugging showed issues arise from:
- `average-entry-valuation` values being corrected incorrectly (being addressed in #6185)
- `virtual-stake` values being updated with the market growth factor incorrectly.

# Observed behaviour

1. When an LP makes a liquidity amendment;  the `average-entry-valuation` values are calculated incorrectly.
3. `virtual-stake` values are being updated with the growth factor `(1+r)` **MORE THAN ONCE** in a market period.

# Expected behaviour

1. When an LP makes a liquidity amendment;  the `average-entry-valuation` values are calculated correctly.
3. `virtual-stake` values must be updated with the growth factor `(1+r)` **ONCE** at the end of a market period.

# Steps to reproduce

## Manual
Steps to reproduce the behaviour manually:
1. Checkout branch `feat/0042-growing-market-feature-test`.
2. Go to `core/integration/features/verified/0042-LIQF-fees_rewards_growing_market.feature`.
3. Read comments to see where the feature test is failing.

# Evidence

Link to [sheet](https://docs.google.com/spreadsheets/d/1ywOsc9QivCzyKE0TNgflD9EQjZ9cXGiSTiW3efEy22w/edit#gid=552695682) with expected values and actual values given by core for Scenario 001. Sheet comments have extra explanations of issues.

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All bug recreation steps can be followed without presenting the original error/bug
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new [system-tests](https://github.com/vegaprotocol/system-tests/issues)
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

### PR DESCRIPTION
**Pull request by EVODelavega**
closes #6063 calculate average entry valuation according to updated spec

### CODE REVIEW

**Review by Jiajia-Cui — DISMISSED** [review]
LGTM

File: core/execution/equity_shares.go [hunk 1]
```diff
@@ -161,27 +158,40 @@ func (es *EquityShares) SetPartyStake(id string, newStakeU *num.Uint) {
 		v.stake = newStake
 		es.totalVStake = es.totalVStake.Add(v.vStake)
 		es.totalPStake = es.totalPStake.Add(v.stake)
-		v.avg = v.vStake.Mul(es.totalPStake.Div(es.totalVStake))
+		// average entry valuation should not be changed when reducing the physical stake
+		// v.avg = v.vStake.Mul(es.totalPStake.Div(es.totalVStake))
```
Comment (jeremyletang) [inline core/execution/equity_shares.go:162]: Is that meand to be here?

File: core/execution/equity_shares.go [hunk 2]
```diff
@@ -161,27 +158,40 @@ func (es *EquityShares) SetPartyStake(id string, newStakeU *num.Uint) {
...
+	es.updateAvgPosDelta(v, delta, newStake)
+	// es.totalVStake = es.totalVStake.Add(delta)
+	// es.totalPStake = es.totalPStake.Sub(v.stake).Add(newStake)
 	// stakes were originally assigned _after_ the mustEquity call
 	// average was calculated in the if es.openingAuctionEnded bit
 	// v.avg = v.vStake.Mul(es.totalPStake.Div(es.totalVStake))
-	v.vStake = v.vStake.Add(delta) // increase
+	// v.vStake = v.vStake.Add(delta) // increase
+	// v.stake = newStake
+	// v.avg = v.vStake.Mul(es.totalPStake.Div(es.totalVStake))
```
Comment (jeremyletang) [inline core/execution/equity_shares.go:174]: ?

File: core/execution/equity_shares.go (hunk 1 above)
Comment (EVODelavega) [reply core/execution/equity_shares.go:162]: Not anymore, I'll remove most stuff I commented out

File: core/execution/equity_shares.go (hunk 2 above)
Comment (EVODelavega) [reply core/execution/equity_shares.go:174]: lots of stuff was moved around to get things to match the spreadsheet and/or the spec file. Going back and forth between things ended up with a lot of lines commented out. Will clean this up

**Review by cdummett — COMMENTED** [review]
Branch is not giving the correct initial AEV values for LP2 in the scenarios in 0042-LIQF-fees_rewards_VirtualStake.feature.

Updated the feature test with comments showing correct values taken from this [sheet](https://docs.google.com/spreadsheets/d/1ywOsc9QivCzyKE0TNgflD9EQjZ9cXGiSTiW3efEy22w/edit#gid=1780176382).

File: core/execution/equity_shares.go [hunk 3]
```diff
@@ -67,8 +67,8 @@ func (es *EquityShares) OpeningAuctionEnded() {
 // for every LP during opening auction.
 func (es *EquityShares) setOpeningAuctionAVG() {
 	// now set average entry valuation for all of them.
-	factor := num.DecimalFromFloat(1.0)
-	if !es.totalVStake.IsZero() && !es.totalPStake.IsZero() {
+	factor := num.DecimalFromFloat(1)
+	if !es.totalPStake.IsZero() && !es.totalVStake.IsZero() {
 		factor = es.totalPStake.Div(es.totalVStake)
```
Comment (cdummett) [inline core/execution/equity_shares.go:72]: Not sure why this is happening? Can't see anything in the spec about updating AEV values at the end of an opening auction?

**Review by jeremyletang — APPROVED** [review]

**Review by cdummett — APPROVED** [review]
LGTM

**Pull request merged**